internal/app/secretless/proxy.go

```diff
@@ -3,49 +3,38 @@ package secretless
 import (
 	"log"
 	"net"
-	"os"
-	"os/signal"
 	"sync"
-	"syscall"
-
 	"github.com/conjurinc/secretless-broker/pkg/secretless/config"
 	plugin_v1 "github.com/conjurinc/secretless-broker/pkg/secretless/plugin/v1"
 )
 
+const (
+	START  = iota
+	RESTART
+	SHUTDOWN
+)
+
 // Proxy is the main struct of Secretless.
 type Proxy struct {
-	Config            config.Config
-	EventNotifier     plugin_v1.EventNotifier
-	Listeners         []plugin_v1.Listener
-	ListenerWaitGroup sync.WaitGroup
-	Resolver          plugin_v1.Resolver
-	RunListenerFunc   func(id string, options plugin_v1.ListenerOptions) plugin_v1.Listener
-	RunHandlerFunc    func(id string, options plugin_v1.HandlerOptions) plugin_v1.Handler
+	Config          config.Config
+	EventNotifier   plugin_v1.EventNotifier
+	Listeners       []plugin_v1.Listener
+	runEventChan    chan int
+	cleanupMutex    sync.Mutex
+	Resolver        plugin_v1.Resolver
+	RunListenerFunc func(id string, options plugin_v1.ListenerOptions) plugin_v1.Listener
+	RunHandlerFunc  func(id string, options plugin_v1.HandlerOptions) plugin_v1.Handler
```

Comment:
Alphabetic order needed - up to you how you want to sort private vs public field groupings.